nuvla/job_engine/connector/nuvlaedge_k8s.py

```diff
@@ -193,23 +193,46 @@ def _get_target_node_name(self):
         return cmd_res.stdout
 
     @staticmethod
-    def _env_to_vars(envs: dict) -> list:
-        """
-        Parse the environment variables and convert them to helm --set vars.
-        """
+    def _get_env_by_pattern(envs: list, pattern: str) -> list:
+        env_conf = []
 
-        env_pair_pattern = r'^\w+=\w+$'
-
-        set_vars = []
         for env_pair in envs:
             log.debug('Environment pair: %s', env_pair)
             env_pair_mod = ''.join(env_pair.split())
-            re_result = re.match(env_pair_pattern, env_pair_mod)
+            re_result = re.match(pattern, env_pair_mod)
             log.debug('Matching result: %s', re_result)
             if re_result:
-                set_vars.extend(['--set ', env_pair_mod])
-                log.debug('Current env vars: %s', set_vars)
+                env_conf.extend(['--set', env_pair_mod])
+                log.debug('Current env vars: %s', env_conf)
+        return env_conf
+
+
+    def _conf_to_vars(self, envs: list) -> list:
```

Comment:
Can you please add a couple of unit tests?

Reply:
Done